sunpy/map/maputils.py

```diff
@@ -102,6 +103,27 @@ def solar_angular_radius(coordinates):
     return np.arctan(coordinates.rsun / coordinates.observer.radius)
 
 
+def sample_at_coords(smap, coordinates):
+    """
+    Samples the data in a map at given series of coordinates.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    coordinates : `~astropy.coordinates.SkyCoord`
+        A list of input coordinates
+
+    Returns
+    -------
+    `~numpy.array`
+        A `numpy.array` corresponding to the data obtained from the map,
+        at the input coordinates.
+    """
+    return smap.data[smap.wcs.world_to_array_index(coordinates)]
```

Comment:
I believe the `coordinates` need to be `SkyCoord`, but since I have not explored the API much as of now, if anyone else can confirm for this by looking into the `wcs`, that would be great.

Reply:
I tested using the `CartesianRepresentation` and the `wcs.world_to_array_index()` gave an error, displaying that it required `SkyCoord` as a parameter.

Reply:
Thanks for checking that.